### PR TITLE
added Cart model and many to many associations

### PR DESCRIPTION
**Pull request by NyjiaT**

### CODE REVIEW

File: server/db/index.js [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
+Cart.belongsTo(User);
+
+// Many to Many
+const cart_product = db.define('cart_product', {
+  quantity: {
+    type: Sequelize.INTEGER
+  }
+});
+
+Cart.belongsToMany(Product, { through: cart_product });
```
Comment (angely119) [inline server/db/index.js:23]: according to the sequelize docs the conventional naming for a through table is camel case so `cartProduct`

File: server/db/index.js [hunk 2]
```diff
@@ -1,18 +1,33 @@
...
+Product.belongsToMany(Cart, { through: cart_product });
 
 module.exports = {
   db,
   models: {
     User,
-    Product
+    Product, 
+    Cart
```
Comment (angely119) [inline server/db/index.js:31]: you will need to use cartProduct to update its quantity later so should probably export it

File: server/db/models/Cart.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+const Sequelize = require('sequelize');
+const db = require('../db');
+
+const Cart = db.define('cart', {})
```
Comment (angely119) [inline server/db/models/Cart.js:4]: need to export

File: server/db/index.js [hunk 4]
```diff
@@ -1,18 +1,33 @@
...
 // Model Associations
-User.belongsToMany(Product, {through: 'Cart'});
-Product.belongsToMany(User, {through: 'Cart'});
+
+// One to One
+User.hasOne(Cart);
+Cart.belongsTo(User);
+
+// Many to Many
+const cart_product = db.define('cart_product', {
```
Comment (angely119) [inline server/db/index.js:17]: according to the sequelize docs the conventional naming for a through table is camel case so cartProduct

and lets move it to its own file models/cartProduct to make it easier to reference later

**Review by angely119 — CHANGES_REQUESTED** [review]
got bug:
Error: user.hasOne called with something that's not a subclass of Sequelize.Model
    at Function.<anonymous> (/Users/angelyang/Documents/grace-shopper/node_modules/sequelize/lib/associations/mixin.js:63:13)

File: server/db/index.js [hunk 5]
```diff
@@ -14,20 +14,21 @@ User.hasOne(Cart);
 Cart.belongsTo(User);
 
 // Many to Many
-const cart_product = db.define('cart_product', {
+const cartProduct = db.define('cartProduct', {
```
Comment (angely119) [inline server/db/index.js:17]: this could be moved to its own file in /models folder

File: server/db/index.js [hunk 6]
```diff
@@ -14,20 +14,21 @@ User.hasOne(Cart);
...
   quantity: {
     type: Sequelize.INTEGER
   }
 });
 
-Cart.belongsToMany(Product, { through: cart_product });
-Product.belongsToMany(Cart, { through: cart_product });
+Cart.belongsToMany(Product, { through: cartProduct });
```
Comment (angely119) [inline server/db/index.js:23]: // Many to Many comment goes above line 23 and 24

File: server/db/index.js (hunk 5 above)
Comment (NyjiaT) [reply server/db/index.js:17]: I'd prefer to keep it in this file atleast for now so its clear that it is being created as a through table but if you really want it in its own file I can refactor it later.

File: server/db/index.js [hunk 7]
```diff
@@ -14,20 +14,21 @@ User.hasOne(Cart);
 Cart.belongsTo(User);
 
 // Many to Many
-const cart_product = db.define('cart_product', {
+const cartProduct = db.define('cartProduct', {
   quantity: {
     type: Sequelize.INTEGER
   }
 });
```
[the remark this answers is not in the document]
Comment (NyjiaT) [reply server/db/index.js:22]: To me defining the through table falls under creating the many to many association along with the actual association statements.

File: client/store/redux/cart.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+const updateQty = (cartProduct, qty) => ({
+  type: UPDATE_QTY,
+  cartProduct,
+  qty
+});
+
+//  Thunks
+
+export const addItemToCart = (cartProduct) => async dispatch => {
+        const { data } = await axios.post(`/api/orders`, cartProduct)
```
Comment (angely119) [inline client/store/redux/cart.js:21]: Before writing your thunk creator, you need a POST /api/orders route that works and returns the correct seed data in postman/insomnia

**Review by angely119 — CHANGES_REQUESTED** [review]
the db/index.js and models/Cart.js files are good to go once you have the changes for the seed file and all the models and associations are reflecting properly on positco.

so for your next commit git add models/Cart.js  db/index.js  and /script/seed.js

**Pull request merged**